cmd/oidc-example-op/server.go

```diff
@@ -0,0 +1,147 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"html/template"
+	"log"
+	"strings"
+	"sync"
+	"time"
+
+	"net/http"
+
+	"github.com/pardot/oidc/core"
+)
+
+const (
+	sessIDCookie = "sessID"
+)
+
+type server struct {
+	oidc     *core.OIDC
+	mux      *http.ServeMux
+	muxSetup sync.Once
+	storage  *storage
+}
+
+const loginPage = `<!DOCTYPE html>
+<html>
+	<head>
+		<meta charset="UTF-8">
+		<title>LOG IN</title>
+	</head>
+	<body>
+		<h1>Log in to IDP</h1>
+		<form action="/finish" method="POST">
+			<p>Subject: <input type="text" name="subject" value="auser" required size="15"></p>
+			<p>Granted Scopes (space delimited): <input type="text" name="scopes" value="{{ .acr }}" size="15"></p>
+			<p>ACR: <input type="text" name="acr" size="15"></p>
+			<p>AMR: <input type="text" name="amr" value="{{ .acr }}" size="15"></p>
+			<p>Userinfo: <textarea name="userinfo" rows="10" cols="30">{"name": "A User"}</textarea></p>
+    		<input type="submit" value="Submit">
+		</form>
+	</body>
+</html>`
+
+var loginTmpl = template.Must(template.New("loginPage").Parse(loginPage))
+
+func (s *server) authorization(w http.ResponseWriter, req *http.Request) {
+	ar, err := s.oidc.StartAuthorization(w, req)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("error starting authorization flow: %v", err), http.StatusInternalServerError)
+		return
+	}
+
+	// set a cookie with the auth ID, so we can track it. Note - this should
+	// _never_ be done in a real application. The auth ID should generally be
+	// kept secret from the user, and the user should not be able to pass one
+	// directly.
```

Comment:
What are the potential issues with exposing the auth ID? I think today we do thread it through via query string parameters, so it is possible to tamper with it.

Reply:
Mainly an opportunity to brute force things, but really that's a problem we have in general. I've removed this statement.